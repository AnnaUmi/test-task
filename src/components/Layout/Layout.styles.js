import styled from 'styled-components';

export const Footer = styled.div`
  height: 100px;
`;

export const Content = styled.div`
  display: flex;
`;
