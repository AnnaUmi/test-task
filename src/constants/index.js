import PropTypes from 'prop-types';

/* eslint-disable import/prefer-default-export */
export const NORMALIZE_URL =
  'https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.0/normalize.min.css';

export const SIDE_NAV_ITEM_TYPES = {
  ListItem: 'ListItem',
  SubMenu: 'SubMenu',
};

export const routerProps = PropTypes.shape({
  pathName: PropTypes.string,
  route: PropTypes.string,
  query: PropTypes.object,
  back: PropTypes.func,
  beforePopState: PropTypes.func,
  prefetch: PropTypes.func,
  push: PropTypes.func,
  reload: PropTypes.func,
  replace: PropTypes.func,
});

export const sideNavSubMenuItemProps = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  url: PropTypes.string,
};

export const sideNavItemProps = {
  ...sideNavSubMenuItemProps,
  type: PropTypes.oneOf([SIDE_NAV_ITEM_TYPES.ListItem, SIDE_NAV_ITEM_TYPES.SubMenu]).isRequired,
};

export const noop = () => {};
