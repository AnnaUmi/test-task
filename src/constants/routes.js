import { SIDE_NAV_ITEM_TYPES } from '.';

export const HEADER_NAV_ROUTES = [
  {
    route: '/',
    label: 'PRISM Logo',
    id: '/',
  },
  {
    route: '/brand',
    label: 'Brand',
    id: '/brand',
  },
  {
    route: '/product',
    label: 'Product',
    id: '/product',
  },
  {
    route: '/marketing',
    label: 'Marketing',
    id: '/marketing',
  },
  {
    route: '/components',
    label: 'Components',
    id: '/components',
  },
];

const componentsSideNav = [
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Buttons',
    id: 'components/buttons',
    url: '/components/buttons',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Typography',
    id: 'components/typography',
    url: '/components/typography',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Grid',
    id: 'components/grid',
    url: '/components/grid',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Spacing',
    id: 'components/spacing',
    url: '/components/spacing',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Colors',
    id: 'components/colors',
    url: '/components/colors',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Breadcrumbs',
    id: 'components/breadcrumbs',
    url: '/components/breadcrumbs',
  },
];

export const SIDE_NAV_ROUTES = {
  '/components': componentsSideNav,
};
