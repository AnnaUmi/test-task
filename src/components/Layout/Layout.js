import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'next/router';
import { routerProps } from '../../constants';
import { Footer, Content } from './Layout.styles';
import HeaderNav from '../HeaderNav';
import SideNav from '../SideNav';
import { HEADER_NAV_ROUTES, SIDE_NAV_ROUTES } from '../../constants/routes';
import { MarkdownContent } from '../MarkdownComponents';

const Layout = ({ router, children }) => {
  const currRoute = router.route;
  const currMainRoute = `/${router.route.split('/')[1]}`;
  const menuItems = SIDE_NAV_ROUTES[currMainRoute];

  return (
    <div>
      <HeaderNav routes={HEADER_NAV_ROUTES} currMainRoute={currMainRoute} />
      <Content>
        {menuItems && <SideNav menuItems={menuItems} currRoute={currRoute} />}
        <MarkdownContent>{children}</MarkdownContent>
      </Content>
      <Footer />
    </div>
  );
};

Layout.propTypes = {
  router: routerProps.isRequired,
  children: PropTypes.node.isRequired,
};

export default withRouter(Layout);
