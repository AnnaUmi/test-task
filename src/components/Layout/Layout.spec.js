import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { noop } from '../../constants';

import Layout from './Layout';

jest.mock('../../constants/routes', () => ({
  HEADER_NAV_ROUTES: [
    {
      route: '/',
      label: 'PRISM Logo',
      id: '/',
    },
    {
      route: '/marketing',
      label: 'Marketing',
      id: '/marketing',
    },
    {
      route: '/components',
      label: 'Components',
      id: '/components',
    },
  ],
  SIDE_NAV_ROUTES: {
    '/components': [
      {
        type: 'ListItem',
        label: 'Buttons',
        id: 'components/buttons',
        url: '/components/buttons',
      },
      {
        type: 'ListItem',
        label: 'Icons',
        id: 'components/icons',
        url: '/components/icons',
      },
    ],
  },
}));

describe('HeaderNav', () => {
  let router;
  let routes;

  beforeEach(() => {
    router = {
      pathName: '/',
      route: '/',
      query: {},
      back: noop,
      beforePopState: noop,
      prefetch: noop,
      push: noop,
      reload: noop,
      replace: noop,
    };

    routes = [
      {
        route: '/components',
        label: 'Components',
        id: '/components',
      },
    ];
  });

  it('should match snapshot when route is root', () => {
    const tree = renderer.create(<Layout router={router}>test children</Layout>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when route is components/buttons', () => {
    router.route = '/components/buttons';
    Layout.routes = routes;

    const tree = renderer.create(<Layout router={router}>test children</Layout>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when route does not exist', () => {
    router.route = '/test';
    Layout.routes = routes;

    const tree = renderer.create(<Layout router={router}>test children</Layout>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
