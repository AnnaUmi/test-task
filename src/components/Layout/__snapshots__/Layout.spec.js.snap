// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav should match snapshot when route does not exist 1`] = `
.c6 {
  height: 100px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

.c5 {
  padding: 0 20px;
  width: 100%;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      PRISM Logo
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        onClick={[Function]}
      >
        <div>
          Marketing
        </div>
      </div>
      <div
        className="c3"
        onClick={[Function]}
      >
        <div>
          Components
        </div>
      </div>
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      test children
    </div>
  </div>
  <div
    className="c6"
  />
</div>
`;

exports[`HeaderNav should match snapshot when route is components/buttons 1`] = `
.c10 {
  height: 100px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

.c4 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4:first-child {
  margin-left: 0;
}

.c4:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: blue;
}

.c4:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: blue;
}

.c6 {
  width: 200px;
  min-height: 200px;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  background-color: #939393;
}

.c7 {
  padding: 20px 30px;
  font-size: 20px;
  cursor: pointer;
  background-color: #7e7e7e;
  font-weight: bold;
}

.c7:hover {
  background-color: #bdbdbd;
}

.c8 {
  padding: 20px 30px;
  font-size: 20px;
  cursor: pointer;
}

.c8:hover {
  background-color: #bdbdbd;
}

.c9 {
  padding: 0 20px;
  width: 100%;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      PRISM Logo
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        onClick={[Function]}
      >
        <div>
          Marketing
        </div>
      </div>
      <div
        className="c4"
        onClick={[Function]}
      >
        <div>
          Components
        </div>
      </div>
    </div>
  </div>
  <div
    className="c5"
  >
    <div
      className="c6"
    >
      <div
        className="c7"
        onClick={[Function]}
      >
        Buttons
      </div>
      <div
        className="c8"
        onClick={[Function]}
      >
        Icons
      </div>
    </div>
    <div
      className="c9"
    >
      test children
    </div>
  </div>
  <div
    className="c10"
  />
</div>
`;

exports[`HeaderNav should match snapshot when route is root 1`] = `
.c6 {
  height: 100px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

.c5 {
  padding: 0 20px;
  width: 100%;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      PRISM Logo
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
        onClick={[Function]}
      >
        <div>
          Marketing
        </div>
      </div>
      <div
        className="c3"
        onClick={[Function]}
      >
        <div>
          Components
        </div>
      </div>
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      test children
    </div>
  </div>
  <div
    className="c6"
  />
</div>
`;
